test/index.js

```diff
@@ -1182,6 +1182,78 @@ describe('list', function() {
 
   });
 
+  describe('take', function() {
+
+    it('returns a Nothing if applied to empty collection', function() {
+      eq(S.take(2, []), S.Nothing());
+      eq(S.take(2, ''), S.Nothing());
+    });
+
+    it('returns a Nothing if n is greater than collection length', function() {
+      eq(S.take(6, ['a', 'b', 'c', 'd', 'e']), S.Nothing());
+      eq(S.take(6, 'abcde'), S.Nothing());
+    });
+
+    it('returns a Nothing if n is less than 0', function() {
+      eq(S.take(-1, ['a', 'b', 'c', 'd', 'e']), S.Nothing());
+      eq(S.take(-1, 'abcde'), S.Nothing());
+    });
```

Comment:
What about `-0`? I would expect this to always result in a Nothing, since we differentiate between positive zero and negative zero in `S.slice`.